internal/application/workload.go

```diff
@@ -353,48 +353,26 @@ func (a *Workload) Restarts(ctx context.Context) (int32, error) {
 	return restarts, nil
 }
 
-// Get returns the state of the app deployment encoded in the workload.
-func (a *Workload) Get(ctx context.Context, deployment *appsv1.Deployment) *models.AppDeployment {
-	active := false
-	stageID := ""
-	status := ""
-	username := ""
-
-	// Query application deployment for stageID and status (ready vs desired replicas)
-
-	deploymentSelector := fmt.Sprintf("app.kubernetes.io/part-of=%s,app.kubernetes.io/name=%s", a.app.Namespace, a.app.Name)
-
-	deploymentListOptions := metav1.ListOptions{
-		LabelSelector: deploymentSelector,
-	}
-
-	deployments, err := a.cluster.Kubectl.AppsV1().Deployments(a.app.Namespace).List(ctx, deploymentListOptions)
+// GetStageID is a specialization of Get coming after, to determine and deliver only the StageId of the workload.
+// Nothing else.
+func (a *Workload) GetStageID(ctx context.Context, deployment *appsv1.Deployment) string {
+	// Query application deployment for stageID
 
-	desiredReplicas := int32(0)
-	readyReplicas := int32(0)
-
-	var createdAt time.Time
+	return deployment.Spec.Template.ObjectMeta.Labels["epinio.suse.org/stage-id"]
+}
 
-	if err != nil {
-		status = pkgerrors.Wrap(err, "failed to get Deployment status").Error()
-	} else if len(deployments.Items) < 1 {
-		status = "0/0"
-	} else {
-		desiredReplicas = deployments.Items[0].Status.Replicas
-		readyReplicas = deployments.Items[0].Status.ReadyReplicas
+// Get returns the state of the app deployment encoded in the workload.
+func (a *Workload) Get(ctx context.Context, deployment *appsv1.Deployment) *models.AppDeployment {
```

Comment:
This cleanup was good but I think we can improve it further. The only consumer or `Get` (https://github.com/epinio/epinio/blob/main/internal/application/application.go#L400-L409), first finds the `Deployment` and then it passes it to `Get`. Can't `Get` simply find the deployment itself by calling `Deployment()` on the Workload object?

This way the signature of this method is simplified: `func (a *Workload) Get(ctx, context.Context) *models.AppDeployment`

Looking at the caller, we should just make sure we still return an `apierrors.IsNotFound(err)` error if a deployment doesn't exist.

Reply:
Now that we have `GetStageID` too, a caller could benefit from having the deployment in params by not having to fetch the deployment twice if both `GetStageID` and `Get` need to be called. Since we don't have a caller like that, we should probably not worry about this. If it's a problem in the future, we can maybe memoize the `Deployment()` result. But given the sideeffects and complication introduce by caching things, I would wait until that is needed.

Reply:
Will look into this after the townhall. ... Might be easier to treat not found as `no error, return nil data`. That is what the caller expects and uses anyway when no deployment is present.

Reply:
Commit 04b00c4a0f149b0e8310143fa38772565f52fed7